tests/std/tests/GH_004388_unordered_meow_operator_equal/test.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <map>
+#include <set>
+#include <unordered_map>
+#include <unordered_set>
+
+using namespace std;
+
+namespace modulo {
+    // Functors for comparison and hashing
+    // based on modular arithmetic equivalence classes
+    template <int N>
+    struct compare {
+        bool operator()(int lhs, int rhs) const {
+            return (lhs % N) < (rhs % N);
+        }
+    };
+
+    template <int N>
+    struct equals {
+        bool operator()(int lhs, int rhs) const {
+            return (lhs % N) == (rhs % N);
+        }
+    };
+
+    template <int N>
+    struct hash {
+        size_t operator()(int value) const {
+            return static_cast<size_t>(value % N);
+        }
+    };
+} // namespace modulo
+
+// Overloaded function to get a key from a set / map's value_type
+int const& get_key(int const& value) {
+    return value;
+}
+int const& get_key(pair<int const, int> const& pair) {
+    return pair.first;
+}
+
+// Equality comparison for unordered sets and maps as per the C++ standard
+// It also gives the correct result for associative containers (though is sub-optimal)
+template <typename Container>
+bool std_equal(Container const& lhs, Container const& rhs) {
+    if (lhs.size() != rhs.size()) {
+        return false;
+    }
+
+    for (auto it = lhs.cbegin(); it != lhs.cend();) {
+        auto const& key = get_key(*it);
+        auto l_range    = lhs.equal_range(key);
+        auto r_range    = rhs.equal_range(key);
+
+        if (!is_permutation(l_range.first, l_range.second, r_range.first, r_range.second)) {
+            return false;
+        }
+        it = l_range.second;
+    }
+    return true;
+}
+
+template <typename Map>
+bool test_maps() {
+    // In all cases, these sets should compare not equal, even though their elements are equivalent
+    Map map1 = {{1, 1}, {2, 2}};
+    Map map2 = {{21, 1}, {12, 2}};
+
+    return (map1 == map2) == std_equal(map1, map2);

```

Comment:
Substantial issue: When writing tests, it's always important to cover both positive and negative cases. This guards against problems, in both the product code and the test code, where a mistake causes the answer to be "always true" or "always false" regardless of the input. Here, we should have test coverage where the containers are actually `==`, to verify that both `operator==` and the test's `std_equal` are getting the right answer.